### DISCUSSION TITLE
Ripgrep+fzf from ADVANCED section tweak question

### DISCUSSION THREAD

**Discussion by iolpltuisciaoln** [junegunn/fzf#3576] [Q&A] [answered]
Hi everyone! I have a question about ripgrep search with fzf from [ADVANCED](https://github.com/junegunn/fzf/blob/master/ADVANCED.md#using-fzf-as-interactive-ripgrep-launcher) section:
```
function rgfzf {
    RG_PREFIX="rg --column --line-number --no-heading --color=always --smart-case "
    INITIAL_QUERY="${*:-}"
    : | fzf --ansi --disabled --query "$INITIAL_QUERY" \
        --bind "start:reload:$RG_PREFIX {q}" \
        --bind "change:reload:sleep 0.1; $RG_PREFIX {q} || true" \
        --bind "alt-enter:unbind(change,alt-enter)+change-prompt(2. fzf> )+enable-search+clear-query" \
        --color "hl:-1:underline,hl+:-1:underline:reverse" \
        --prompt '1. ripgrep> ' \
        --delimiter : \
        --preview 'bat --color=always {1} --highlight-line {2}' \
        --preview-window 'up,60%,border-bottom,+{2}+3/3,~3' \
        --bind 'enter:become(vim {1} +{2})'
}
```

Although  the function itself is very impressive i'd like to tweak it a little bit, particularly i'm finding a way to postpone beginning of search until i start typing search pattern in Ripgrep section.
Any ideas how to achieve it? 
Thanks.

**Comment by LangLangBart** [comment, marked as answer]
>  a way to postpone beginning of search until i start typing search pattern in Ripgrep section

To do this, remove the following line from the function:

```bash
  --bind "start:reload:$RG_PREFIX {q}"
```

You can find more details under the `AVAILABLE EVENTS:` section in the `man fzf` page.
